ios/ThreadsDevSettings.m

```diff
@@ -0,0 +1,37 @@
+#import "ThreadsDevSettings.h"
+#import <React/RCTLog.h>
+
+@implementation ThreadsDevSettings
+// We're replacing the stock implementation of RCTDevSettings for the worker bridge.
+// We define `moduleName` instead of using RCT_EXPORT_MODULE because the latter would
+// override RCTDevSettings for the parent bridge, which is undesirable.
++ (NSString *)moduleName
+{
+  return @"RCTDevSettings";
+}
+
+// RCTDevSettings doesn't expose requiresMainQueueSetup, so we explicitly keep it in sync.
++ (BOOL)requiresMainQueueSetup
+{
+  return YES;
+}
+
+- (BOOL)isShakeToShowDevMenuEnabled
+{
+  return NO;
+}
+
+- (BOOL)isRemoteDebuggingAvailable
+{
+  return NO;
+}
+
+- (void)_remoteDebugSettingDidChange
+{
+  if (super.isDebuggingRemotely && !self.isRemoteDebuggingAvailable) {
+    return;
```

Comment:
This prevents the debugger from running on the thread.

An issue I am facing right now is that the thread's bridge still init-s `RCTDevMenu`, which subscribes to keyboard shortcuts. And since the thread inits after the main app this means we open the `DevMenu` for the thread. Ideally we would disable the thread's `DevMenu`, but I haven't been able to achieve this. Replacing the `RCTDevMenu` in a similar fashion as I do for `RCTDevSettings` leads to missing modules errors. 

You can still test the `DevSettings` approach works correctly though by enabling debugging before you run with this PR and observing debugging still works after.

@mvayngrib any ideas here for `DevMenu`?

Reply:
hm, i tried changing the code in RCTDevMenu to use the below, but then i see Remote JS Debugger Unavailable in the menu. Maybe it'll give u an idea?
```obj-c
- (RCTDevMenu *)devMenu
{
#if RCT_DEV
  return [[self.parentBridge.bundleURL absoluteString] containsString:@"index.thread"] ? nil : [self moduleForClass:[RCTDevMenu class]];
#else
  return nil;
#endif

}
```

Reply:
could another option be to have a static bundle for the thread vs served over http? Maybe that'll disable debugging

Reply:
> hm, i tried changing the code in RCTDevMenu to use the below, but then i see Remote JS Debugger Unavailable in the menu. Maybe it'll give u an idea?
> 
> ```objc
> - (RCTDevMenu *)devMenu
> {
> #if RCT_DEV
>   return [[self.parentBridge.bundleURL absoluteString] containsString:@"index.thread"] ? nil : [self moduleForClass:[RCTDevMenu class]];
> #else
>   return nil;
> #endif
> 
> }
> ```

Thx, will take a look at mutating RN code as well

Reply:
> could another option be to have a static bundle for the thread (like in prod) vs served over http? Maybe that'll disable debugging. The assumption is that the thread code doesn't change often

Yeah, that should be possible, but would require the dev to bundle the app first before running and could cause weird behaviour if threads code changes (though I agree this should not happen often)

Reply:
>  but would require the dev to bundle the app first before running 

not if we add it to git :) But i have a feeling people won't like this

> and could cause weird behaviour if threads code changes (though I agree this should not happen often)

yes and yes

Reply:
> ```objc
>   return [[self.parentBridge.bundleURL absoluteString] containsString:@"index.thread"] ? nil : [self moduleForClass:[RCTDevMenu class]];
> ```

Just tested and even after this the thread `DevMenu` still opens (hence the `Remote JS Debugger Unavailable`). Tested by logging `bridgeURL` in `RCTDevMenu.toggle`.

Will keep looking

Reply:
Figured it out, I can stop the listeners from registering based on the bundle URL 🥳

Reply:
Thx for the help